test/unit/services/test_system.py

```diff
@@ -2,284 +2,374 @@
-    """
+    def check_create_success(self):
+        """Checks that a prior call to `call_create` worked as expected"""
 
-    system_id = str(ObjectId())
-    system = MagicMock()
+        self.wrapped_utils.generate_code.assert_called_once_with(self._expected_system_out.name, "system")
+        self.mock_system_repository.create.assert_called_once_with(self._expected_system_in)
 
-    # Mock `get` to return a System
-    test_helpers.mock_get(system_repository_mock, system)
+        assert self._created_system == self._expected_system_out
 
-    retrieved_system = system_service.get(system_id)
 
-    system_repository_mock.get.assert_called_once_with(system_id)
-    assert retrieved_system == system
+class TestCreate(CreateDSL):
+    """Tests for creating a System"""
 
+    def test_create(self):
+        """Test creating a System"""
 
-def test_get_with_non_existent_id(test_helpers, system_repository_mock, system_service):
-    """
-    Test getting a System with a non-existent ID
+        self.mock_create(SYSTEM_POST_DATA_NO_PARENT_A)
+        self.call_create()
+        self.check_create_success()
 
-    Verify that the `get` method properly handles the retrieval of a System with a non-existent ID
-    """
-    system_id = str(ObjectId())
+    def test_create_with_parent_id(self):
+        """Test creating a System with a parent ID"""
 
-    # Mock `get` to not return a System
-    test_helpers.mock_get(system_repository_mock, None)
+        self.mock_create({**SYSTEM_POST_DATA_NO_PARENT_A, "parent_id": str(ObjectId())})
+        self.call_create()
+        self.check_create_success()
 
-    retrieved_system = system_service.get(system_id)
 
-    system_repository_mock.get.assert_called_once_with(system_id)
-    assert retrieved_system is None
+class GetDSL(SystemServiceDSL):
+    """Base class for get tests"""
 
+    _obtained_system_id: str
+    _expected_system: MagicMock
+    _obtained_system: MagicMock
 
-def test_get_breadcrumbs(test_helpers, system_repository_mock, system_service):
-    """
-    Test getting breadcrumbs for a system
+    def mock_get(self):
+        """Mocks repo methods appropriately to test the 'get' service method"""
 
-    Verify that the `get_breadcrumbs` method properly handles the retrieval of a System
-    """
-    system_id = str(ObjectId())
-    breadcrumbs = MagicMock()
+        # Simply a return currently, so no need to use actual data
+        self._expected_system = MagicMock()
+        self.test_helpers.mock_get(self.mock_system_repository, self._expected_system)
 
-    # Mock `get` to return breadcrumbs
-    test_helpers.mock_get_breadcrumbs(system_repository_mock, breadcrumbs)
+    def call_get(self, system_id: str):
+        """Calls the SystemService `get` method"""
 
-    retrieved_breadcrumbs = system_service.get_breadcrumbs(system_id)
+        self._obtained_system_id = system_id
+        self._obtained_system = self.system_service.get(system_id)
```

Comment:
It's complaining about the type - `Expected type 'MagicMock', got 'SystemOut | None' instead `.